source/common.h

```diff
@@ -880,8 +880,8 @@ class cmdline_processor
         -> void
     {
         help_requested = true;
-        std::string_view a = __DATE__;
-        std::string_view b = __TIME__;
+        const std::string_view a = __DATE__;
+        const std::string_view b = __TIME__;
         std::unordered_map<std::string_view, char> m = { {"Jan",'1'}, {"Feb",'2'}, {"Mar",'3'}, {"Apr",'4'}, {"May",'5'}, {"Jun",'6'}, {"Jul",'7'}, {"Aug",'8'}, {"Sep",'9'}, {"Oct",'A'}, {"Nov",'B'}, {"Dec",'C'} };
```

Comment:
`m` also smells like it cold be `const`?

Reply:
It might, but `operator[]` is not const in `std::unordered_map` (https://en.cppreference.com/w/cpp/container/unordered_map/operator_at).

`m` might be a const array with custom lookup... but I perceive it as premature optimization as the impact of this should be first measured.

Reply:
Definitely a premature optimization for a function that is only ever called during command line processing, and thus likely only once per process.  If anything, this would be a pessimization in general because it might cause the compiler to create this at compile time and thus increase the size of the binary unnecessarily.